### ISSUE TITLE
Improved default tab shortcuts

### ISSUE THREAD

**Issue by gilbertohasnofb** [frescobaldi/frescobaldi#1335]
Most programs (e.g. Firefox, Chromium, Nemo) seem to use the same commands for switching between tabs: `Ctrl+Tab` for next tab, `Ctrl+Shift+Tab` for previous tab, and `Alt+<number>` for the n-th tab. Could you consider implementing the same shortcuts by default?

Using `Ctrl+Tab` for next document works on my version of Frescobaldi (2.18.1), but `Ctrl+Shift+Tab` for previous document does not work. Also, I can't seem to find any shortcut to go for a specific tab.

**Comment by fedelibre** [comment]
> Using `Ctrl+Tab` for next document works on my version of Frescobaldi (2.18.1), but `Ctrl+Shift+Tab` for previous document does not work. Also, I can't seem to find any shortcut to go for a specific tab.

By default the shortcuts to go to next previous tabs are `Alt + right arrow` and `Alt + left arrow`. So what is working or not working for you is simply set in the Preferences>Shortcuts menu.

That said, I agree that `Alt+<number>` would be useful.
Right now it's used by some default snippets to set the voice number.

**Comment by gilbertohasnofb** [comment]
Hi, thanks for the quick reply and apologies for the lack of clarity in the
original message. This issue has actually three parts:

- suggesting the default bindings for next and previous tab to be changed
from Alt+left/right to Ctrl+Tab/Ctrl+Shift+Tab as to match the convention
used in most software
- suggest Alt+number to go to a specific tab, as is conventional as well
- report that overriding the previous document shortcut cannot be changed
into Ctrl+Shift+Tab. The new binding is accepted but the shortcut doesn't
work, so this might be a bug. Overriding the next document shortcut to
Ctrl+Tab does work though.

**Comment by fedelibre** [comment]
By default Ctrl+Tab is used to move the focus between the panels (this is quite common in other applications I use). In fact when you try to add the shortcut to another action it says there's a conflict with "next frame".

Ctrl+Shift+Tab by default does nothing AFAICS. I can set it as a shortcut, so I cannot reproduce your problem.

For tab navigation I see some apps using `Ctrl+PageDown` to move to the right and `Ctrl+PageUp` to move to the left. For example, [Firefox 80](https://support.mozilla.org/en-US/kb/keyboard-shortcuts-perform-firefox-tasks-quickly#w_windows-tabs) and VS Code 1.49 on Linux.

Personally, I like the Alt+arrows shortcuts. It's much easier and quicker than any alternative. Frescobaldi is not a browser, so why trying to emulate it?

+1 for the Alt+number shortcut.

**Comment by gilbertohasnofb** [comment]
> By default Ctrl+Tab is used to move the focus between the panels (this is quite common in other applications I use). In fact when you try to add the shortcut to another action it says there's a conflict with "next frame".

Indeed. I also got the warning about the conflict, then I chose to override it anyway and the shortcut works, i.e. `Ctrl+Tab` does cycle through tabs.

> Ctrl+Shift+Tab by default does nothing AFAICS. I can set it as a shortcut, so I cannot reproduce your problem.

`Ctrl+Shift+Tab` does nothing by default over here either, and I can set it as a shortcut without a problem. The issue is that the new shortcut does _not_ work, i.e. pressing `Ctrl+Shift+Tab` does not cycle backwards the tabs. Are you able to press this combination and cycle backwards through the tabs in your system after setting it as the preferred shortcut?

> For tab navigation I see some apps using Ctrl+PageDown to move to the right and Ctrl+PageUp to move to the left. 
> [...]
> Personally, I like the Alt+arrows shortcuts. It's much easier and quicker than any alternative. Frescobaldi is not a browser, so why trying to emulate it?

You are right about `Ctrl+PageUp` and `Ctrl+PageDown`, that seems to be one of the most common default shortcuts in other text editors. I was not trying to suggest emulating a browser, I just suggested considering a more commonly used shortcut. Personally, I don't think I ever came across `Alt+arrows` for tab navigation, which is why I mentioned it, that's all.

**Comment by fedelibre** [comment]
> `Ctrl+Shift+Tab` does nothing by default over here either, and I can set it as a shortcut without a problem. The issue is that the new shortcut does _not_ work, i.e. pressing `Ctrl+Shift+Tab` does not cycle backwards the tabs. Are you able to press this combination and cycle backwards through the tabs in your system after setting it as the preferred shortcut?

Yes, it's  working fine here (Fedora 33, Frescobaldi 3.1.2).

**Comment by fedelibre** [comment]
> You are right about `Ctrl+PageUp` and `Ctrl+PageDown`, that seems to be one of the most common default shortcuts in other text editors. I was not trying to suggest emulating a browser, I just suggested considering a more commonly used shortcut. Personally, I don't think I ever came across `Alt+arrows` for tab navigation, which is why I mentioned it, that's all.

Yep, I see your point :-)
You give priority to the "common use", while I give priority to "ease of use".

Alt+arrows is quicker than Ctrl-Tab and Ctrl-Shift-Tab (3 keys together!). I think we can agree here.
But I don't want to convince you :-) 
Fortunately, Frescobaldi is configurable.
